esmvaltool/diag_scripts/gwls/plot_gwl_exceedance_mm_stats.py

```diff
@@ -0,0 +1,252 @@
+    ProvenanceLogger,
+    io,
+    run_diagnostic,
+    select_metadata,
+)
+
+SEP = "_"
+
+logger = logging.getLogger(Path(__file__).stem)
+
+
+def log_provenance(provenance, filename, cfg):
+    """Create a provenance record for the output file."""
+    with ProvenanceLogger(cfg) as provenance_logger:
+        provenance_logger.log(filename, provenance)
+
+
+def calculate_gwl_mm_cube(
+    project_data,
+    gwl_subset_df,
+    window_size,
+):
+    """Calculate multimodel stats around GWL exceedance years."""
+    cubes = iris.cube.CubeList()
+
+    datasets = gwl_subset_df["Model"].unique()
+    for dataset in datasets:
+        # get every exp for which the data is available for
+        all_exps_in_dataset = select_metadata(project_data, dataset=dataset)
+        for el in all_exps_in_dataset:
+            cube = iris.load_cube(el["filename"])
+            logger.info("Cube var name is %s", cube.var_name)
+            if cube.var_name == "tas":
+                cube.convert_units("Celsius")
+            if cube.var_name == "pr":
+                cube = cube * 86400
+                cube.units = "mm/day"
+
+            ih.prepare_cube_for_merging(cube, el["filename"])
+            # extract window period
+            year_of_exceedance = gwl_subset_df[
+                (gwl_subset_df["Model"] == dataset)
+                & (gwl_subset_df["Exp"] == el["exp"])
+            ]["Exceedance_Year"].values[0]
+            if np.isnan(year_of_exceedance):
+                continue
+            start_year = int(year_of_exceedance - ((window_size - 1) / 2))
+            end_year = int(year_of_exceedance + ((window_size - 1) / 2))
+            logger.info(
+                "Model: %s, Exp : %s start year: %s, endyear: %s",
+                dataset,
+                el["exp"],
+                str(start_year),
+                str(end_year),
+            )
+            # Start year Jan 1st (1, 1) and end_year Dec 31st (12, 31)
+            cube = extract_time(cube, start_year, 1, 1, end_year, 12, 31)
+            cube = cube.collapsed("time", iris.analysis.MEAN)
+            cubes.append(cube)
+    return cubes
+
+
+def calculate_mm_stats(cubes, mean_file, stdev_file):
+    """Calculate mean and standard deviation from merged mm cubes."""
+    # find index of time coord so this can be unified across data sets
+    # before merging as the time data points are different across models/scenarios
+    index = 0
+    for coord in cubes[0].aux_coords:
+        if coord.standard_name == "time":
+            break
+        index = index + 1
+    for c in cubes[1:]:
+        c.remove_coord("time")
+        c.add_aux_coord(cubes[0].aux_coords[index])
+
+    if len(cubes) == 0:
+        logger.info("No model instances exceed this GWL.")
+    elif len(cubes) == 1:
+        iris.save(cubes[0], mean_file)
+        logger.info("No standard deviation calculated for a single instance.")
+    else:
+        mm_cube = cubes.merge_cube()
```

Comment:
`merge_cube` is always fussy, maybe encase in a try/except with a relevant exception that prints the cubes?

Reply:
I have added a try/exception statement for this now.